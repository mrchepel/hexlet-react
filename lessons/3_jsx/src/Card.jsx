/* eslint-disable react/display-name */
import React from 'react';

const getCard = ({ title, text }) => {
  if (!title && !text) {
    return null;
  }
  const vdom = (
    <div className="card">
      <div className="card-body">
        {title && <h4 className="card-title">{title}</h4>}
        {text && <p className="card-text">{text}</p>}
      </div>
    </div>
  );
  return vdom;
};

export default getCard;
