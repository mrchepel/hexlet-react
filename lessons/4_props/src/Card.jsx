/* eslint-disable react/prefer-stateless-function */

import React from 'react';
import PropTypes from 'prop-types';

export default class Card extends React.Component {
  render() {
    const { title, text } = this.props;
    return (
      <div className="card">
        <div className="card-body">
          <h4 className="card-title">{title}</h4>
          <p className="card-text">{text}</p>
        </div>
      </div>
    );
  }
}
Card.propTypes = {
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};
