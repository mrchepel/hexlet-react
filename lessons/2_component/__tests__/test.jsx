import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../src/Card';

test('Card', () => {
  const component = renderer.create(<Card />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
