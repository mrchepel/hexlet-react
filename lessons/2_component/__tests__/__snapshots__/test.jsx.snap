// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <h4
      className="card-title"
    >
      Card title
    </h4>
    <p
      className="card-text"
    >
      Some quick example text to build on the card
    </p>
    <button
      className="btn btn-primary"
      type="button"
    >
      Go somewhere
    </button>
  </div>
</div>
`;
